[Ruth] iP

ruthpohrp opened the pull request:
# Super Pretty Ultra Hyper Mega Cute Shining God of Destruction Dukuwu ~~☆☆☆ tehe
> dowcto, wewre losing him!!
> qwick, hand me the defibwiwatow!!
> please. turn off my [redacted] life support
>                              - You after you use Dukuwu [(source)](https://i.redd.it/qcsnneo5fyh01.jpg)

## What Dukuwu Does To You
Dukuwu sucks your will to live. It's:
* Time consuming
* Sanity consuming
* Flesh consuming

Sounds fun, right? And the best part is that it's **F R E E !**
All you need to do are these three simple steps:
1. Download it [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ&ab_channel=RickAstley)
2. If you actually clicked the link, lmao ok no the actual one is [here](https://github.com/ruthpohrp/ip)
3. You ~~don't actually~~ want to download this ~~_WHY WOULD YOU_~~
4. please. turn off my life support

Features:
- [x] A detailed narration of my slow descent into my insanity
- [x] Nothing else
- [x] Oh yeah and Dukuwu helps you manage your tasks ig 🤔 just run the Duke file and type 'help' to find out more

## 
If you Java noob, Google is your best friend. Take it from me. Here's a sneak peek at what it looks like:
```javascript
public static void run(Storage storage, TaskList tasklist) throws DukeException {
        String logo = " ____        _        \n"
                + "|  _ \\ _   _| | _____ \n"
                + "| | | | | | | |/ / _ \\\n"
                + "| |_| | |_| |   <  __/\n"
                + "|____/ \\__,_|_|\\_\\___|\n";
        String linebreakstart = "~~*********___\\(owo)/___\\(owo)/___*********~~\n";
        String linebreakend = "~~*********___\\(owo)/___\\(owo)/___*********~~";

        System.out.println(linebreakstart
                + "\nHewwo!! From:\n"
                + logo
                + "How mayw Iw hewlp youw, Mastwer? Pwease type 'help' for a list of commandws uwu\n");
```

Commit:
Add Gradle support

Commit:
Level-1

Commit:
Level-2

Commit:
Level-3

Commit:
Level-3 added further check if integer after done is valid

Commit:
Level-4 added Deadline (incomplete), Task (edited), Todo (complete)

Commit:
Level-4 Completed Deadline. Colours are starting to blur before my eyes.

Commit:
Level-4. Now with Deadline (no assumptions that /by is inside) support.

Commit:
Level-4. Done as far as I can tell. May need revision but I'm tired

Commit:
A-TextUiTesting check. Done.

Commit:
Level-5. To add the other handling errors when I have time

Commit:
Level-5

Commit:
Level-5. Improved text and deleted stuff that made no sense.

Commit:
Merge branch 'master' of https://github.com/ruthpohrp/ip

# Conflicts:
#	src/main/java/Duke.java

Commit:
Level-5. Changed text so read smoother

Commit:
Level-5 edited. Added handling for done.

Commit:
Level-6. Improved text and added delete function + handling.

Commit:
Tweaks handling errors some more. Made it better.

Commit:
Added support for done function (if its already marked as done), and cleaned up wording of exception

Commit:
Added javadocs. Makes it easier to read ig

Commit:
to start on Level-7

Commit:
progression of adding dataeditor

Commit:
added tasklist for easier reference. argh

Commit:
modified tasklist and more of level-7

Commit:
done level-7. whew

Commit:
adds date time parses for level-8

Commit:
changed to include localdate. level-8

Commit:
added date support but not time (yet). level-8

Commit:
Merge branch 'branch-Level-7'

* branch-Level-7:
  done level-7. whew
  modified tasklist and more of level-7
  added tasklist for easier reference. argh
  progression of adding dataeditor

Commit:
Merge branch 'branch-Level-8'

* branch-Level-8:
  added date support but not time (yet). level-8
  changed to include localdate. level-8
  adds date time parses for level-8

Commit:
added ui, changed name of storage, parser

Commit:
updated parser, used more enum.

Commit:
ok this time the enum in ui should work properly, plus added try catch on while loop to be neater so it can catch parse error

Commit:
moved to package duke

Commit:
changed small spacing

Commit:
added test and jar.

Commit:
added javadocs and coding standards

Commit:
added javadoc, coding standard

Commit:
added support for level-9.

Commit:
reformatted testing and parser. tasklist now only adds tasks instead of doing parsing as well.

Commit:
added more parsetests.

Commit:
added support for time in input (optional)

Commit:
solved some parsing-saving mistakes.

leeroy999 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,38 @@
+package duke;
+
+import java.lang.reflect.Array;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
```
Might want to check your imports to clean up any unnecessary imports.

ruthpohrp replied:
Done, thanks!

leeroy999 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,38 @@
+        new Duke("data/tasks.txt").run();
+    }
+
+}
```
It would be a good style to put new line as a last character for formatting.

ruthpohrp replied:
Done, thanks

leeroy999 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,38 @@
+
+    public static void main(String[] args) throws DukeException {
+        new Duke("data/tasks.txt").run();
+    }
```
Although these methods are quite intuitive, it would be good practice to add header comments (Javadocs) for these public methods.

ruthpohrp replied:
Oops done thanks

leeroy999 commented on src/main/java/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,3 @@
+Manifest-Version: 1.0
+Main-Class: duke.Duke
+
```
I think this file is not necessary to be committed.

ruthpohrp replied:
Oh.... why not? I thought it was important

leeroy999 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,109 @@
+package duke;
```
It would be better to package the tasks related files in a task package.

ruthpohrp replied:
Ah yes done thanks

leeroy999 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,116 @@
+/**
+ * Occurs when Duke doesn't recognise the input.
+ */
+class InvalidInputException extends DukeException {
+    InvalidInputException() {
+        super ("Sowwy, thiws commandw iswn't supporwted! TwT");
+    }
+}
```
Each class should have their own separate file. Even though the method has default access modifier, it would be good practice to add header comments.

You can also consider putting all the exceptions in the same package.

There should not be a space after `super`
(same comment for other classes in this file)

ruthpohrp replied:
Done, thanks!

leeroy999 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,106 @@
+
+    /**
+     * Returns time Event occurs at.
+     * @return
```
Header could be "Getter method for time Event occurs", and the return description could be what you have written in you header, instead of being empty.

ruthpohrp replied:
Oops done thanks

leeroy999 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,106 @@
+
+    /**
+     * Returns simplified date and time Event occurs at, in String form.
+     * @return
```
Similar to previous, return should include a description, instead of being in the header.

ruthpohrp replied:
Yup done thanks

leeroy999 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,188 @@
+     * @param strparse Array of Strings to be joined together.
+     * @return Keyword to be found.
+     */
+    public String parseFind(String[] strparse) {
```
```suggestion
    public String parseFind(String[] strParse) {
```

ruthpohrp replied:
WHOOPS, done thanks

leeroy999 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,188 @@
+     * @return Todo task.
+     * @throws DukeException Errors that occur during parsing (incorrect commands, etc.).
+     */
+    public Todo parseTodo(String[] strparse) throws DukeException {
```
Same as above, variables should have camelCase. Likewise for strparse in this file

ruthpohrp replied:
Replaced all, thanks!

leeroy999 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,188 @@
+     * @return Keyword to be found.
+     */
+    public String parseFind(String[] strparse) {
+        StringBuilder strb = new StringBuilder();
```
Variables with large scope should have long names. A suggestion would be to change `strb` to `builder` which could improve readability. Same as other StringBuilders.

ruthpohrp replied:
It's not a global variable though? So I don't need to empty the StringBuilder each time. I'll replace it with strBuilder to improve readability like you said, thanks!

Commit:
changed visibility

leeroy999 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,188 @@
+        int minute = time.getMinute();
+        String hourInString;
+        String minuteInString;
+        String AmOrPm;
```
```suggestion
        String amOrPm;
```

ruthpohrp replied:
Oops replaced thanks

leeroy999 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,275 @@
+ * @author Ruth Poh
+ */
+public class TaskList {
+    ArrayList<Task> taskArr;
```
Plural form should be used on names representing a collection of objects. For example: tasks

ruthpohrp replied:
Done, thanks

leeroy999 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,275 @@
+     * @return Number of tasks in TaskList that is done.
+     * @throws DukeException Occurs when anything goes wrong during method.
+     */
+    public int markDone(String[] strparse) throws DukeException {
```
```suggestion
    public int markDone(String[] strParse) throws DukeException {
```

leeroy999 replied:
Same for other strparse in this file, and other files

ruthpohrp replied:
Done, thanks

leeroy999 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,275 @@
+        } catch (NumberFormatException e) {
+            throw new MissingNoException("done");
+        }
+        int i = Integer.parseInt(strparse[1]);
```
Variables with a large scope should have long names.

ruthpohrp replied:
Yup done thanks

leeroy999 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,275 @@
+     * @param i Number of task in TaskList that is done.
+     * @throws DukeException Occurs when anything goes wrong during method.
+     */
+    public void markReadDone(int i) throws DukeException {
```
Variables with a large scope should have long names. Same as other `int i` in this file and other files which are declared in large scope.

ruthpohrp replied:
Done, may need to check other areas, thanks

leeroy999 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,275 @@
+     * @throws DukeException Occurs when anything goes wrong during method.
+     */
+    public String displayList() throws DukeException {
+        StringBuilder strb = new StringBuilder();
```
Variables with a large scope should have long names. Same for other StringBuilder variables in this file and other files.

ruthpohrp replied:
Done, thanks

leeroy999 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,220 @@
+        System.out.println(linebreakend);
+
+        boolean continueLoop = true;
+        // for checking loop
```
Minor problem, but comments should be written on the line(s) before the code. Applies to the rest of the comments

ruthpohrp replied:
Done, thanks!

leeroy999 commented on src/test/java/duke/TaskListTest.java:
```diff
@@ -0,0 +1,44 @@
+        tasklist.add(new Event("Revive me", LocalDate.of(2022, 12,12)));
+        tasklist.add(new Event("Revive me", LocalDate.of(2022, 12,12),
+                LocalTime.parse("23:23")));
+        TaskList t = new TaskList(tasklist);
```
Variables with a large scope should have long names.

ruthpohrp replied:
Done, thanks!

leeroy999 reviewed (commented):
Overall lgtm, good job! Just have to take note a few nits in coding standards styling. In general, it would be good to refrain from copying code, such as in `Task.java` where there are many `return "";` lines. It could be an abstract class as well. But well done for iP in this short amount of time :)

Commit:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d'

* commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d':
  Add Gradle support

Commit:
added gradle support, starting on javafx

Commit:
new branch for level 10

Commit:
added files for GUI support

Commit:
cleaned up ui and duke.

Commit:
launcher works now but now it won't quit when i send a bye

Commit:
'bye' now works thanks to kamisama vimuth. todo: add command support and run() support.

Commit:
Merge branch 'branch-Level-10'

* branch-Level-10:
  'bye' now works thanks to kamisama vimuth. todo: add command support and run() support.
  launcher works now but now it won't quit when i send a bye
  cleaned up ui and duke.
  added files for GUI support
  new branch for level 10

Commit:
build.grade file: compatible for generating jar with gradle

Commit:
Parser class: Assert for parserDeadline and parserEvent that Deadline and Event contain /by and /at keywords respectively

Deadline and Event need to have /by and /at keywords to separate their task and time of deadline or event.

Let's add an assertion to signal something is wrong with the parse checking since it means an IncorrectInputException is not being returned.

Assertion is being used in addition to the exception check in case there is a missing check for the exception.

Commit:
Exception classes in an exceptions folder and Tasks classes in a tasks folder in main/java/duke

The Exception classes should have their own file instead of being lumped into one file.

Task and Exception classes number more than three and make the duke folder look messy.

Let's put them in packages named tasks and exceptions for the according classes.

Other files outside of the packages will import files from these packages aka duke.exceptions and duke.tasks.

Note that some names of Exceptions/Errors have been changed as well to reflect their type of error/exception.

Commit:
Duke, Storage, Tasklist, Exceptions classes have improved code quality and javadocs support

After shifting tasks and exceptions to their respective packages, there is now a discrepancy in throwing errors.

Javadocs for the respective classes are also either too general or outright missing.

Let's clean that up by importing the correct packages and write specific javadocs for the classes.

Commit:
New and improved javadocs and code quality

Comments on Github pointed out my noob mistakes and suggested
much needed improvements to code quality.

Let's read the comments and fix the code accordingly.

Many thanks to leeroy999. My code is even sexier(?) now.

Commit:
New and improved images for user and duke in the GUI

ruthpohrp committed:
Merge pull request #1 from ruthpohrp/branch-A-Assertions

Add assertions to Parser class

Commit:
Merge branch 'master' into branch-A-CodeQuality

* master:

ruthpohrp committed:
Merge pull request #2 from ruthpohrp/branch-A-CodeQuality

New and improved code quality of all duke classes

Commit:
Merge branch 'master' of https://github.com/ruthpohrp/ip

* 'master' of https://github.com/ruthpohrp/ip:
  New and improved images for user and duke in the GUI
  New and improved javadocs and code quality
  Duke, Storage, Tasklist, Exceptions classes have improved code quality and javadocs support
  Exception classes in an exceptions folder and Tasks classes in a tasks folder in main/java/duke

Commit:
Improved GUI

Commit:
Improve GUI and Code Quality

Commit:
Improved GUI and Code Quality.

Changes to GUI:
* Smaller profile picture (Easter Egg in DialogBox.fxml: lol)
* Luciana Grande font, and bigger font size
* Better alignment of text in Dialog Boxes

Commit:
Improved CodeQuality in Task, Deadline, Todo, Event

Commit:
Improved Code Quality (an ongoing process...)

Commit:
Improved CodeQuality (an ongoing process) and updated README.md

ruthpohrp committed:
Set theme jekyll-theme-minimal

Commit:
Added Ui.png

Commit:
Merge commit '02728dcdb36f0edf16786d1f17818bc35d5a4420'

* commit '02728dcdb36f0edf16786d1f17818bc35d5a4420':
  Set theme jekyll-theme-minimal

ruthpohrp committed:
Update README.md

ruthpohrp committed:
Update README.md

Commit:
Update docs/README.md (ongoing)

Commit:
Update README.md. The process continues

Commit:
Updated README.md

Commit:
Update docs/README.md

Commit:
Updated website theme at https://ruthpohrp.github.io/ip/

ruthpohrp committed:
Set theme jekyll-theme-minimal

Commit:
Updated Ui to promote the website, need to figure out how to execute jar by double clicking

Commit:
Dukewu.jar now works when done from command line, but not from double clicking

Commit:
Updated both README.md and docs/README.md

Commit:
Command classes added. Ui is now more streamlined.

Commit:
Fixed a mistake where time is not shown in 'list' command
when Storage reads from tasks.txt.

Commit:
Updated 'clearall' command boolean. It now works and requires confirmation.
Also, clearing an empty task list will result in an error message.
Shame on you.

Commit:
Easter Egg

Commit:
Easter Egg in README.md

Commit:
Updated README.md to include website details

Commit:
MANIFEST.MF reflects version number

ruthpohrp committed:
Event.png deleted

.png not loading so deleted.

ruthpohrp committed:
Event.png added

Hopefully it works now

ruthpohrp committed:
Added Event.png

Manually so it should refer in README.md

ruthpohrp committed:
Set theme jekyll-theme-cayman

ruthpohrp committed:
Set theme jekyll-theme-minimal

ruthpohrp committed:
Add CmdSummary.png

ruthpohrp committed:
Update README.md

man

ruthpohrp committed:
Update README.md (table)

hopefully this works
